version: '3.8'

networks:
  challange_network:
    ipam:
      config:
        - subnet: 172.28.0.0/24

services:
  app1:
    tty: true
    build:
      context: .
      dockerfile: ./.docker/python/Dockerfile
    container_name: flask_api
    networks:
      challange_network:
        ipv4_address: 172.28.0.2
    ports:
      - "5000:5000"
  database:
    image: mysql:8.0
    restart: always
    environment:
      - MYSQL_DATABASE=challange
      - MYSQL_ROOT_PASSWORD=abcd1234
    networks:
      challange_network:
        ipv4_address: 172.28.0.3
    ports:
      - "3360:3306"
    volumes:
      - database:/var/lib/mysql

volumes:
  database:
    driver: local
